ably-ios/ARTRealtimeChannel.m

```diff
@@ -148,6 +148,26 @@ - (void)publishProtocolMessage:(ARTProtocolMessage *)pm cb:(ARTStatusCallback)cb
     }
 }
 
+
+- (void)sendMessage:(ARTProtocolMessage *)pm cb:(ARTStatusCallback)cb {
+    __block BOOL gotFailure = false;
+    __block ARTEventListener *listener = [self.realtime.connection on:^(ARTConnectionStateChange *stateChange) {
```

Comment:
What then? The spec says "If a connection enters the CLOSED or FAILED state [...] the client should consider the delivery of those messages as failed", and that's what's happening here. (Maybe this is being done at the wrong level, though, because RTN7 is about ACK and NACK...)